### ISSUE TITLE
SCIM LDAP sync fails due to broken check on sync owned entries

### ISSUE THREAD

**Issue by ultinous-kszegi** [kanidm/kanidm#3100]
### I did this

Tried to run LDAP sync

### I expected the following

That my nonexistent cat give birth to puppies. Just kidding. LDAP sync runs.

### This happened instead

sync fails

Excerpt from server log, debug level
```
scim_sync_apply_phase_3 [ 2.69ms | 28.33% ]
┝━ 🐛 [debug]: Schemas valid - Proceeding with entry <entry-id>
┝━ 🐛 [debug]:  | requested_classes: {"account": SchemaClass { name: "account", uuid:0000-0000-0000-0000-ffff00000046, description: "Object representation of an account", sync_allowed: true, systemmay: [AccountExpire, AccountValidFrom, NameHistory], may: [], systemmust: [DisplayName, Name, Spn], must: [], systemsupplements: ["oauth2_resource_server", "person", "service_account"], supplements: [], systemexcludes: [], excludes: [] }, "person": SchemaClass { name: "person", uuid: 00000000-0000-0000-0000-ffff00000044, description: "Object representation of a person", sync_allowed: true, systemmay: [ApplicationPassword, AttestedPasskeys, CredentialUpdateIntentToken, LegalName, Mail, OAuth2ConsentScopeMap, OAuth2Session, PassKeys, PrimaryCredential, RadiusSecret, SshPublicKey, UserAuthTokenSession], may: [], systemmust: [IdVerificationEcKey], must: [], systemsupplements: [], supplements: [], systemexcludes: ["application", "service_account"], excludes: [] }}
┝━ 🐛 [debug]:  | sync_owned_attrs: {"account_expire", "account_valid_from", "attested_passkeys", "displayname", "legalname", "mail", "name", "name_history", "password_import", "primary_credential", "radius_secret", "ssh_publickey", "totp_import", "unix_password_import"}
┕━ 🚨 [error]: Rejecting attribute displayName for entry 502d69d6-ea34-103c-887c-4d0a4203daaf which is not sync owned
 ```

Excerpt from the output of `kanidm system sync get <syncaccount>` (all `sync_yield_authority` values)
```
sync_yield_authority: passkeys
sync_yield_authority: phone
sync_yield_authority: sshpublickeys
```

### Kanidm version details
nushell FTW
```
❯ zypper se -is kanidm | from zypper
╭───┬────┬────────────────┬─────────┬───────────────────────────────────┬────────┬───────────────────────────╮
│ # │ s  │      name      │  type   │              version              │  arch  │        repository         │
├───┼────┼────────────────┼─────────┼───────────────────────────────────┼────────┼───────────────────────────┤
│ 0 │ i+ │ kanidm-clients │ package │ 1.4.0~dev~git75.cf63c6b-lp156.1.1 │ x86_64 │ [Home]: firstyear: kanidm │
│ 1 │ i+ │ kanidm-server  │ package │ 1.4.0~dev~git75.cf63c6b-lp156.1.1 │ x86_64 │ [Home]: firstyear: kanidm │
╰───┴────┴────────────────┴─────────┴───────────────────────────────────┴────────┴───────────────────────────╯

```

**Comment by Firstyear** [comment]
Can you add the ldap sync client config?

**Comment by ultinous-kszegi** [comment]
Of course 
```
❯ open /etc/kanidm/ldap-sync.toml | reject ldap_ca ldap_sync_pw sync_token
╭────────────────────────┬───────────────────────────────────────────────────╮                                                                                                            
│ group_attr_description │ cn                                                │                                                                                                            │
│ group_attr_name        │ uid                                               │                                                                                                            │
│ ldap_filter            │ (|(objectclass=person)(objectclass=groupOfNames)) │                                                                                                            │ ldap_sync_base_dn      │ dc=example.com,dc=examplehosting                  │                                                                                                            
│ ldap_sync_dn           │ cn=kanidm,ou=services,dc=common,dc=examplehosting │                                                                                                            
│ ldap_uri               │ ldaps://ldap.example.com                          │                                                                                                            
│ max_ber_size           │ 1310720                                           │                                                                                                            │
│ person_attr_mail       │ mail,mailAlias                                    │                                                                                                            │
│ person_attr_password   │ userpassword                                      │                                                                                                           
│ person_attr_user_name  │ uid                                               │                                                                                                            │
╰────────────────────────┴───────────────────────────────────────────────────╯  
```

**Comment by ultinous-kszegi** [comment]
Could you reopen this?

```
🐛 [debug]:  | sync_owned_attrs: {AccountExpire, AccountValidFrom, AttestedPasskeys, DisplayName, LegalName, Mail, Name, NameHistory, PasswordImport, PrimaryCredential, RadiusSecret, SshPublicKey, TotpImport, UnixPasswordImport}
 🚨 [error]: Rejecting attribute passwordImport for entry 502d69d6-ea34-103c-887c-4d0a4203daaf which is not sync owned
```

passwordImport sneaked through the attribute name normalization

**Comment by Firstyear** [comment]
@ultinous-kszegi I will investigate again this week. we are about to do the release branch for 1.4, but any fixes we make will be backported to 1.4.
